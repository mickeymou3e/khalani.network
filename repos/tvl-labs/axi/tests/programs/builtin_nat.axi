!prove_by_eval(1 add 2 = 3);
!prove_by_eval(10 lt (2 mul 8));
!prove_by_eval(17 le ((2 mul 8) add 1));

# Div axioms.
!claim(
    forall y: Nat, x: Nat. (
        (Zero lt y & y le x) ==>
        x div y = Succ((x sub y) div y)
    )
);

!claim(
    forall y: Nat, x: Nat. (
        ~(Zero lt y & y le x) ==>
        x div y = Zero
    )
);
