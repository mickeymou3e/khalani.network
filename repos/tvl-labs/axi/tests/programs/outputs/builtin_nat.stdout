Result: Proved(=[add[1, 2]):Nat, 3])
Result: Proved(lt[10, mul[2, 8]):Nat])
Result: Proved(le[17, add[mul[2, 8]):Nat, 1]):Nat])
Result: Proved(forall Nat. forall Nat. if(and(lt[Zero:Nat, @1:Nat], le[@1:Nat, @0:Nat]), =[div[@0:Nat, @1:Nat]):Nat, Succ[div[sub[@0:Nat, @1:Nat]):Nat, @1:Nat]):Nat]):Nat]))
Result: Proved(forall Nat. forall Nat. if(not(and(lt[Zero:Nat, @1:Nat], le[@1:Nat, @0:Nat])), =[div[@0:Nat, @1:Nat]):Nat, Zero:Nat]))
